Ubuntu lxde

sonnguyenxg opened the pull request:
Script build ubuntu-lxde for RZ-SBC

sonnguyenxg committed:
ubuntu-lxde: Init source code add file Readme and Main script

Changes include:
	- Add README.md
	- Add main script

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: prepare ubuntu base

Changes include:
	- Add script to prepare ubuntu 22.04 base.
	- Call function ubuntu_base_prepare in main script.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Add prepare_rootfs_qt

Changes include:
	- Tar file rootfs qt
	- Copy kernel, kernel module, overlays, uEnv from core-image-qt
to ubuntu.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Prepare config file

Changes include:
	- Copy quemu binary
	- Copy log file, network interface, lightdm config.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Add script mount chroot to install package

Changes include:
	- Add script mount chroot.
	- Add script run in chroot mode to install basic software.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Set password for root user

Change include:
	- Set password for root user

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Create normal user

Change include:
	- Add normal user: rzpi
	- Enable permission sudo, audio, video.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Add script to install audio and video

Changes include:
	- Add script to install audio and video.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Install wifi and bluetooth

Changes include:
	- Add script to install firmware wifi and application.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Package rootfs to bz2 file

Change include:
	- Call function package rootfs to make bz2 file.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Install lxde-desktop

Change include:
	- Add script to install lxde-desktop.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Update network_interfaces to default only enable wifi

Change include:
	- Update file network_interfaces to default only enable wifi

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Add config camera

Change include:
	- Add script and function to auto enable camera after boot.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Update Audio Video Wifi bluetooth firmware

Change include:
	- Update to install audio video wifi bluetooth firmware withdow show dialog.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Update lxde desktop to install withdow show dialog

Change include:
	- Update lxde desktop to install auto withdow show dialog.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Install chrominium and virtual keyboard

Changes include:
	- Install chrominium-browser and virtual keyboard.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Update function package_rootfs to remove old rootfs

Change include:
	- Check and remove old rootfs if exist.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Add function copy kernel modules

Change include:
	- Add function copy kernel modules from core-image-qt to ubuntu.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

vudangRVC commented:
@sonnguyenxg ,
As your PR is the same as @long-trinh-rvc's [PR#2](https://github.com/vudangRVC/rz-build-scripts-sst/pull/2)
Please check my comments on Long's PR and update this PR

sonnguyenxg committed:
ubuntu-lxde: add script to set permission user root and rzpi

Changes include:
	- Add script to set permission user root and rzpi.
	- Add comment in main function.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Add sync after umount

Changes include:
	- Add sync after umount.
	- Remove some un-use line of code.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Copy wifi firmware and download bluetooth firmware

Changes include:
	- Add function copy wifi firmware from rootfs_qt.
	- Add function download bluetooth firmware.
	- Clear some lines of code.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxde: Create swap partition

Changes include:
	- Add function script to create swap partition and auto enable
swap.
	- Delete function install chromium.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg committed:
ubuntu-lxed: Add detailed comments and descriptions for better clarity

Changes include:
	- Add detailed comments in all script.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>

sonnguyenxg commented:
Hello @vudangRVC ,
I updated script, 
Please help me review it. 
Thanks.

vudangRVC commented:
@sonnguyenxg ,

Please check this file from Ubuntu-core PR: [config.ini](https://github.com/vudangRVC/rz-build-scripts-sst/pull/2/files#diff-ef8ffb58292b682e17a860c201b6feb803be2b371b4856fa4838b854dced4d21)
And create a corresponding file for Ubuntu-lxde. We shall not hardcode any settings in the script.

sonnguyenxg committed:
ubuntu-lxde: Add NetworkManager configuration and file copy function.

Changes include:
	- Add NetworkManager configuration file to enable network by
GUI.
	- Add function copy config file.

Signed-off-by: Son Nguyen <son.nguyen.xg@bp.renesas.com>
